mm2src/coins/eth/v2_activation.rs

```diff
@@ -169,10 +191,18 @@ pub async fn eth_coin_from_conf_and_request_v2(
     ticker: &str,
     conf: &Json,
     req: EthActivationV2Request,
-    priv_key_policy: PrivKeyBuildPolicy,
+    priv_key_policy: EthPrivKeyBuildPolicy,
 ) -> MmResult<EthCoin, EthActivationV2Error> {
-    if req.nodes.is_empty() {
-        return Err(EthActivationV2Error::AtLeastOneNodeRequired.into());
+    let is_metamask = priv_key_policy.is_metamask();
+    let nodes_specified = req.nodes.is_empty();
+
+    if is_metamask && !nodes_specified {
```

Comment:
It's quite interesting that Metamask is both `priv_key_policy` and also RPC mode. Can this be work around by an enum? These conditions become error-prone quite frequently.

Reply:
Yes agree, will add `mode: EthRpcMode` as it's done for UTXO coins

Reply:
Added `rpc_mode: "Metamask"` to the request.
This refactoring allowed also to fix the following bug - previously, `EthCoin::web3_instances` was empty if `priv_key_policy: "Metamask"`. Now everything works correctly.
Thanks!